### ISSUE TITLE
Refactoring

### ISSUE THREAD

**Issue by Serg-Norseman** [Serg-Norseman/Arbor#15]
1. Walk through the source code using ReSharper, to rename the part of variables and other names for compliance with conventions. 
2. To perform the standard VS code analysis. 
3. To check code metrics VS.

**Comment by ruslangaripov** [comment]
> to rename ... for compliance with conventions

With what "conventions"?
:smiling_imp:

**Comment by Serg-Norseman** [comment]
It is mirage, in my mind )))))

**Comment by Serg-Norseman** [comment]
Learn to use issues and labels. Alas, found out too late that the reference at issue properly should be put in the commit message.

**Comment by Serg-Norseman** [comment]
I like C++ and Java style of naming methods lowerCamelCase() (i.e. obj.verbNoun()). But ReSharper handles the methods along with properties and public fields of classes. Bummer... I.e. change in ReSharper rule for one and it changes the processing of two categories

**Comment by ruslangaripov** [comment]
I got it! That's a good idea! May be after that "decoding" the code will become more easier for me! :smiling_imp:

**Comment by Serg-Norseman** [comment]
Let's make a text document with conventions in the project. And gradually average our common preferences. mmm... For sample. I don't like "m_fieldName" or "_fieldName" (I like format 'fFieldName'), but if you like only it - I can accept it. Etc.

**Comment by ruslangaripov** [comment]
That's a good idea too. Yes, such document is a vital one. "Coding conventions" is a very necessary document. And we can try to create it, undoubtedly. But this is a very tough task :laughing: In any case both of us will have to make a trade-off.

Yes, I do prefer `m_member` syntax for data members. Yes I do like operators overloading, **small** C++ lambdas and do not like big ones. I do not like when an opening curly brace is placed on one line with the controlling statement :smiling_imp: I do not like when declaration of data members is placed on the top of class declaration. I like to group class' members by their access modifiers. And so on and so forth :smiling_imp: 

Many my style/coding preferences are driven by "C++ mentality". There's no doubt. As your ones, I believe, are driven by your favorites programming languages.

While we craft the code in different "branches" using different programming languages we can safely omit the "coding conventions". But as soon as we start to work with the same source code the document will become a vital part of the work.

I propose the following stuff: let's begin to create "Coding conventions" document. And because currently we are working with different parts of the source code we have a time to adjust the document. It will be very hard but I believe we can reach agreements :smiling_imp: G'luck :laughing:

If the document will envelop many projects (not only "Arbor") the document must be created in the separated dedicated repository. Should I create it? :smiling_imp:

**Comment by Serg-Norseman** [comment]
You will be probably surprised, but my preferences were formed out of a terrible mixture of languages and sources. I.e. I just chose what like it is to me and my "sense of beauty". On the tradition of a certain language, be it C#, Delphi or Java I scored. So my style is not like my job nor our C# developers or VB developers. More precisely they did not like what I was working alone and for yourself. In collaborative projects I always adhered to a common style.
